packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/common.rs

```diff
@@ -32,7 +32,7 @@ pub type AttributeName = &'static str;
 ///
 /// Each component type added to `ComponentEnum` must implement that component node traits.
 
-#[derive(Debug, EnumString)]
+#[derive(Debug, EnumString, Display)]
```

Comment:
It looks like the `strum::Display` macro is just being used for error messages. We already have the `.get_component_type()` method that gives the component name (as it would be used in `xml`), and that is used in other error messages. We should be consistent.

Maybe implement display for `ComponentEnum` that uses `get_componentType()`? Then, to maintain consistency, we could remove the reference to `.get_component_type()` in `state_var_updates.rs#226`.

It would take more work to get rid of `.get_component_type()` in `components/common.rs#249`, below, as that would require component developers to implement `Display` to call `.get_component_type()` on their struct, which is probably not worth the tiny gain of consistency.

Reply:
I think we need to discuss this a little more. Since the xml name of the tag is not a one-to-one match of which component is being rendered. E.g., a `text` inside of a graph vs. outside. What do we want displayed in the error message? Something for the user or something for the developer, or both...?

Reply:
Though, with `tsify` handling the errors, it is now moot.

Reply:
Right, I was thinking of these errors being displayed in the document, so they would be for the user. Hence, the desire for having the error message show the xml name that the author would have typed in. We can come back to this as we further develop the errors.